app/notification.go

```diff
@@ -258,7 +260,7 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 				// add new followers to existing followers
 				if threadMembership.Following && !followers.Contains(userID) {
```

Comment:
Please expand the mutex critical section to include the entire if condition. That should fix the race. I don't believe there is a need to introduce an additional map.